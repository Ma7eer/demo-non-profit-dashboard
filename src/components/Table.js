import React from 'react'
import PropTypes from 'prop-types'

import '../App.css'

// Import React Table
import ReactTable from 'react-table'
import 'react-table/react-table.css'

const Table = ({
  tableData,
  columnData,
  title,
  defaultPageSize = 10,
  loading,
  r
}) => {
  const data = tableData
  let columns = columnData
  return (
    <>
      <h1>{title}</h1>
      <ReactTable
        data={data}
        columns={columns}
        defaultPageSize={defaultPageSize}
        filterable
        loading={loading}
        className="-striped -highlight"
        style={{ backgroundColor: 'white', color: '#292f33', fontSize: '18px' }}
        ref={r}
        noDataText="لا يوجد بيانات"
        defaultFilterMethod={(filter, row) =>
          String(row[filter.id])
            .toLowerCase()
            .includes(filter.value.toLowerCase())
        }
        getTheadGroupProps={() => {
          return {
            style: {
              background: 'linear-gradient(to bottom, #25aae1, #18a4e0)', //'#16a085',#19b394, #50c9ac
              color: 'white'
            }
          }
        }}
        getTheadProps={() => {
          return {
            style: {
              background: 'linear-gradient(to top, #110c11, #616161)', //'#110c11',
              color: 'white'
            }
          }
        }}
        getTheadThProps={() => {
          return {
            style: {
              border: '0.5px solid #e0dfdf'
            }
          }
        }}
        getTdProps={() => {
          return {
            style: {
              border: '0.1px solid #110c11'
            }
          }
        }}
      />
    </>
  )
}

Table.propTypes = {
  tableData: PropTypes.array,
  columnData: PropTypes.array,
  title: PropTypes.string,
  defaultPageSize: PropTypes.number,
  loading: PropTypes.bool,
  r: PropTypes.any
}

export default Table
