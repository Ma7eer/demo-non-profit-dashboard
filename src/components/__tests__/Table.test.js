import '@testing-library/react/cleanup-after-each'
import React from 'react'
import { render } from '@testing-library/react'
import Table from '../Table'
import ReactTable from 'react-table'

const data = [
  {
    name: 'Tanner Linsley',
    age: 26,
    friend: {
      name: 'Jason Maurer',
      age: 23
    }
  }
]

const columns = [
  {
    Header: 'Name',
    accessor: 'name' // String-based value accessors!
  },
  {
    Header: 'Age',
    accessor: 'age',
    Cell: props => <span className="number">{props.value}</span> // Custom cell components!
  },
  {
    Header: props => <span>Friend Age</span>, // Custom header components!
    accessor: 'friend.age'
  }
]

test('Table renders only title when table and column data is empty', () => {
  const { container } = render(
    <Table tableData={[]} columnData={[]} title={'test title'} />
  )
  expect(container.firstChild).toMatchSnapshot()
})

test('Table renders only title and table with column', () => {
  const { container } = render(
    <>
      <h1>{'title'}</h1>
      <ReactTable
        data={data}
        columns={columns}
        defaultPageSize={10}
        filterable
        loading={false}
        className="-striped -highlight"
        style={{ backgroundColor: 'white', color: '#292f33', fontSize: '18px' }}
        noDataText="لا يوجد بيانات"
        defaultFilterMethod={(filter, row) =>
          String(row[filter.id])
            .toLowerCase()
            .includes(filter.value.toLowerCase())
        }
      />
    </>
  )
  expect(container).toMatchSnapshot()
})
