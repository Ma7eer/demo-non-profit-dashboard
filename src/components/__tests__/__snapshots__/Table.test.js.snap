// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders only title and table with column 1`] = `
<div>
  <h1>
    title
  </h1>
  <div
    class="ReactTable -striped -highlight"
    style="background-color: white; color: rgb(41, 47, 51); font-size: 18px;"
  >
    <div
      class="rt-table"
      role="grid"
    >
      <div
        class="rt-thead -header"
        style="min-width: 300px;"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Name
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Age
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              <span>
                Friend Age
              </span>
            </div>
            <div
              class="rt-resizer"
            />
          </div>
        </div>
      </div>
      <div
        class="rt-thead -filters"
        style="min-width: 300px;"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <input
              style="width: 100%;"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <input
              style="width: 100%;"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <input
              style="width: 100%;"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="rt-tbody"
        style="min-width: 300px;"
      >
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              Tanner Linsley
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span
                class="number"
              >
                26
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              23
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination-bottom"
    >
      <div
        class="-pagination"
      >
        <div
          class="-previous"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          class="-center"
        >
          <span
            class="-pageInfo"
          >
            Page
             
            <div
              class="-pageJump"
            >
              <input
                aria-label="jump to page"
                type="number"
                value="1"
              />
            </div>
             
            of
             
            <span
              class="-totalPages"
            >
              1
            </span>
          </span>
          <span
            class="select-wrap -pageSizeOptions"
          >
            <select
              aria-label="rows per page"
            >
              <option
                value="5"
              >
                5 rows
              </option>
              <option
                value="10"
              >
                10 rows
              </option>
              <option
                value="20"
              >
                20 rows
              </option>
              <option
                value="25"
              >
                25 rows
              </option>
              <option
                value="50"
              >
                50 rows
              </option>
              <option
                value="100"
              >
                100 rows
              </option>
            </select>
          </span>
        </div>
        <div
          class="-next"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      class="-loading"
    >
      <div
        class="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;

exports[`Table renders only title when table and column data is empty 1`] = `
<h1>
  test title
</h1>
`;
